staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -748,27 +752,87 @@ func (s *store) GetList(ctx context.Context, key string, opts storage.ListOption
 			growSlice(v, 2048, len(getResp.Kvs))
 		}
 
+		// needs to be large enough to give value in terms of parallelizing work
```

Comment:
Changing all Lists to use at minimum one extra goroutine feels undesirable.

Also, this is a lot of code to add to a function.  Is there an easier to read way to factor this?

Reply:
There’s also minimal value in this when a kms isn’t in use?  Non kms clusters get slower / have more gosched pressure?

Any benchmarks showing how this impacts very large lists both with and without kms (benchmark is probably required for this to merge with a variety of factors considered)?